src/test/java/org/humancellatlas/ingest/security/AccountServiceTest.java

```diff
@@ -0,0 +1,61 @@
+package org.humancellatlas.ingest.security;
+
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {DefaultAccountService.class})
+public class AccountServiceTest {
+
+    @Autowired
+    private AccountService accountService;
+
+    @MockBean
+    private AccountRepository accountRepository;
+
+    @Nested
+    @DisplayName("Registration")
+    class Registration {
+
+        @Test
+        void success() {
+            //given:
+            String providerReference = "67fe90";
+            Account account = new Account(providerReference);
+            assumeThat(account.getRoles()).isEmpty();
```

Comment:
The assumption makes it explicit that that current requirement is that Account are construtcted with no roles. When we change the rules, this test will fail, prompting us to make appropriate changes in the code (if we need to).

Reply:
> The assumption makes it explicit that that current requirement is that Accounts are constructed with no roles.

But GuestAccounts are constructed with non-empty roles (they contain GUEST as their role), and aren't GuestAccounts always passed into the register() method since they are the only account that has access to the /register endpoint?

Reply:
That's for the `GuestAccount`. This is an assumption for the generic Account that was created in the test. I guess we need another test that removes the GUEST role in the service.

Reply:
A Guest account is-an Account, constructed with non-empty roles. What's the required behaviour when a GuestAccount is supplied to register()? 

Since Accounts and GuestAccounts can be mutated with Roles, the requirement "current requirement is that Accounts are construtcted with no roles" isn't guaranteed. Indeed, AccountService.register() can be called with a GuestAccount and mutated providerRef/subjectId and then a further mutated Account with the Contributor and Guest role will be persisted and returned.

It sounds more like register() should take something resembling UserInfo as a parameter. Then the requirement that accounts aren't assigned roles prior to registration is already encoded in the model (this scenario becomes impossible).